capstone-frontend/src/components/Browse.jsx

```diff
@@ -1,10 +1,103 @@
-import React from 'react';
-import BookSearchList from './BookSearchList';
+import React, { Component } from 'react';
+import { Nav, Row, Spinner, Tab } from 'react-bootstrap';
+import { BookSearchList } from './BookSearchList';
+import { ClubSearchList } from './ClubSearchList';
 
-const Browse = (props) => {
-  return (
-    <BookSearchList searchQuery={props.searchQuery} bookLists={props.bookLists} updateBookLists={props.updateBookLists} />
-  );
-}
+import '../styles/Browse.css';
 
-export default Browse;
\ No newline at end of file
+export class Browse extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchedBooks: [],
+      loading: true, // Spinner
+    }
+  }
+
+  getSearchedBooks = (searchQuery) => {
+    fetch(`/api/search?searchTerm=${searchQuery}`)
+      .then(response => response.json())
+      .then(searchedBooks => this._isMounted && this.setState({ searchedBooks, loading: false }))
```

Comment:
This looks like it should work, but I would spend a little time trying to figure out why the component is un-mounting with an outstanding network request if you haven't already.

Reply:
This un-mounting only takes place if the user is trying to navigate away from the search page less than a second after they execute the query. It is uncommon for it to happen where a user is using the application regularly, but I could not figure out a more reliable fix at the moment.